### PR TITLE
Adds arm wrestling

### PR DESCRIPTION
**Pull request by Ryll-Ryll**
## About The Pull Request
Have you ever wanted to settle an important debate with someone in a physical display of power, without actually beating (or slashing/shooting/etc...) the snot out of each other? Do you want to prove your superiority over the weaklings that make up your coworkers? Is the bartender becoming very insistent about finding out who exactly is going to be paying for all the drinks your party has had?

Put away those dueling pistols and say no more! Now you can arm wrestle! See below for an example.

https://cdn.discordapp.com/attachments/326831214667235328/752736837398954084/arm_wrassle.mp4

Simply emote "*awrassle" to pull out your arm and set it on a table to declare a challenge. Anyone who picks up the arm item will be entered into an arm wrestling contest with you, where you'll both constantly deal each other stamina damage with modifiers for your physical traits (this still needs fleshing out, but right now being a giant is good and a dwarf is bad) until one of you falls over and loses. It's as easy as that!

I intentionally chose to leave the contest as non-interactive other than the traits you have at the start (so no DDR minigame or button mashing) to avoid issues with rate limiting/autoclickers/loudness of button mashing/complexity of coding said minigames. I am still looking to flesh out the traits that can affect the contest to add some more depth, as well as add more visuals and flair to the contest to make it more fun to watch. That said, the basic system itself is fully functional (as long as you don't have hulks involved, since they can't take stamina damage which means they're invincible in this). Also, the ability to cheat by doping with stamina regenerating drugs is fully intentional, if you're sneaky enough to pull something like that.

## Why It's Good For The Game
Whimsy, lets people have honor duels over things without the long, drawn out process of boxing duels.

## Changelog
:cl: Ryll/Shaps
add: You can now arm-wrestle with other people! Simply emote "*awrassle" then place the arm item on a table, and have someone else click on it! Assert your dominance over your coworkers!
/:cl:

### CODE REVIEW

**Comment by MrDoomBringer** [comment]
participant sprites should animate/wobble back and forth with increasing frequency until the victor wins

**Comment by Ryll-Ryll** [comment]
Taking any and all suggestions to raise the hype factor/immersion of the contest's visuals

**Comment by itseasytosee** [comment]
surplus prosthetics should suck and cyborg arms should dominate.

**Comment by itseasytosee** [comment]
Also if it goes on for long enough, do to drugs or hulks or whatever, the table should start taking damage eventually breaking and ending the arm wrestle.

**Comment by MrDoomBringer** [comment]
@itseasytosee idk, i feel like having a 5 minute arm wrestle between two methed-up hulks would be pretty funny

**Comment by cacogen** [comment]
Depending on factors that affect strength and arm quality/integrity you should be able to dismember their arm, resulting in it being transferred to the hand slot you were using to arm wrestle.

**Comment by Jared-Fogle** [comment]
There should be some sort of animation, like tilting and shaking or something.

File: code/datums/components/arm_wrassle.dm [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+	var/mob/living/carbon/owner = parent
+	switch(owner.get_held_index_of_item(arm_item))
+		if(LEFT_HANDS)
+			owner_arm = owner.get_bodypart(BODY_ZONE_L_ARM)
+		if(RIGHT_HANDS)
+			owner_arm = owner.get_bodypart(BODY_ZONE_R_ARM)
+		else
+			qdel(src)
+			return
```
Comment (Jared-Fogle) [inline code/datums/components/arm_wrassle.dm:48]:
```suggestion
			return INITIALIZE_HINT_QDEL
```

File: code/datums/components/arm_wrassle.dm [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+	RegisterSignal(arm_item, COMSIG_MOVABLE_MOVED, .proc/check_moved_arm)
+	RegisterSignal(arm_item, COMSIG_ATOM_ATTACK_HAND, .proc/check_grab_correct_hand)
+	RegisterSignal(arm_item, COMSIG_ITEM_EQUIPPED, .proc/check_arm_equipped)
+	RegisterSignal(arm_item, COMSIG_PARENT_QDELETING, .proc/void)
+
+	RegisterSignal(arena, COMSIG_PARENT_QDELETING, .proc/void)
+	RegisterSignal(owner_arm, COMSIG_PARENT_QDELETING, .proc/void)
+	RegisterSignal(challenger_arm, COMSIG_PARENT_QDELETING, .proc/void)
+
+	owner.visible_message("<span class='notice'>[owner] sets [owner.p_their()] [owner_arm.name] on \the [arena], awaiting a challenger!</span>")
```
Comment (Jared-Fogle) [inline code/datums/components/arm_wrassle.dm:59]:
```suggestion
	owner.visible_message("<span class='notice'>[owner] sets [owner.p_their()] [owner_arm.name] on [arena], awaiting a challenger!</span>")
```

I believe this is unnecessary.

File: code/datums/components/arm_wrassle.dm [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+	owner.transferItemToLoc(arm_item, arena.drop_location(), force=TRUE, silent = FALSE)
```
Comment (Jared-Fogle) [inline code/datums/components/arm_wrassle.dm:60]:
```suggestion
	owner.transferItemToLoc(arm_item, arena.drop_location(), force = TRUE, silent = FALSE)
```

Consistency.

File: code/datums/components/arm_wrassle.dm [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+
+/datum/component/arm_wrassle/UnregisterFromParent()
+	UnregisterSignal(parent, list(COMSIG_MOVABLE_MOVED, COMSIG_PARENT_QDELETING))
+
+/// Someone picked up the arm item. If it was the owner, we cancel the challenge, otherwise if it's a carbon (which it should be) we start the contest
+/datum/component/arm_wrassle/proc/check_arm_equipped(obj/item/source, mob/user, slot, initial)
+	SIGNAL_HANDLER
+	var/mob/living/carbon/owner = parent
+	if(user == owner)
+		owner.visible_message("<span class='notice'>[owner] removes [owner.p_their()] [owner_arm.name] from \the [arena].</span>")
```
Comment (Jared-Fogle) [inline code/datums/components/arm_wrassle.dm:83]:
```suggestion
		owner.visible_message("<span class='notice'>[owner] removes [owner.p_their()] [owner_arm.name] from [arena].</span>")
```

File: code/datums/components/arm_wrassle.dm [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+/// A possible challenger has clicked on the arm_item with an empty hand, check if they're using the correct sided hand
+/datum/component/arm_wrassle/proc/check_grab_correct_hand(obj/item/source, mob/living/carbon/potential_challenger)
+	SIGNAL_HANDLER
+
+	if(!istype(potential_challenger))
+		return
+
+	var/mob/living/carbon/owner = parent
+	if(potential_challenger.active_hand_index != owner_arm.held_index)
+		potential_challenger.visible_message("<span class='warning'>[potential_challenger] tries grabbing [owner]'s [owner_arm.name] with the wrong hand!</span>")
```
Comment (Jared-Fogle) [inline code/datums/components/arm_wrassle.dm:104]: This is hillarious.

File: code/datums/components/arm_wrassle.dm [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+		return COMPONENT_NO_ATTACK_HAND // idiot
+	return
+
```
Comment (Jared-Fogle) [inline code/datums/components/arm_wrassle.dm:107]: Not necessary at the end of a proc.

File: code/datums/components/arm_wrassle.dm [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+			owner_arm.throw_at(arm_item.loc)
+		qdel(src)
+
+/// The owner is allowed to move while making the challenge, as long as they're still adjacent to the table the arm_item is on
+/datum/component/arm_wrassle/proc/check_owner_move(atom/movable/mover, atom/oldloc, direction)
+	SIGNAL_HANDLER
+	var/mob/living/carbon/owner = parent
+	if(!owner.Adjacent(arm_item))
+		if(stage > AWRASSLE_STAGE_INITIALIZING && arena)
+			owner.visible_message("<span class='notice'>[owner] walks away from \the [arena].</span>")
```
Comment (Jared-Fogle) [inline code/datums/components/arm_wrassle.dm:183]:
```suggestion
			owner.visible_message("<span class='notice'>[owner] walks away from [arena].</span>")
```

**Comment by AdamElTablawy** [comment]
Being a hulk and the other guy not being a hulk should lead to you slamming their arm through the table, breaking the table and dislocating their arm.

**Comment by Qustinnus** [comment]
wtf you're the wounds guy but didnt add the epic bone snaps that happen during arm wrestling??

**Comment by RaveRadbury** [comment]
Maybe certain chems could aid/hinder the wrassler? Doping scandal!

File: code/game/objects/items/weaponry.dm [hunk 8]
```diff
@@ -936,3 +936,31 @@ for further reading, please see: https://github.com/tgstation/tgstation/pull/301
...
+	var/challenge_set = FALSE
+
+/obj/item/arm_wrassle/dropped(mob/user, silent)
+	if(!challenge_set)
+		qdel(src)
+		return
+	return ..()
+
+/obj/item/arm_wrassle/attack_obj(obj/O, mob/living/user)
+	if(!istype(O, /obj/structure/table))
```
Comment (ATH1909) [inline code/game/objects/items/weaponry.dm:962]: how does this interact with rolling tables?

**Comment by ATH1909** [comment]
> Maybe certain chems could aid/hinder the wrassler? Doping scandal!

Stamina regen chems (including the humble epinephrine) already do just that.

File: code/datums/components/arm_wrassle.dm [hunk 9]
```diff
@@ -0,0 +1,189 @@
...
+	if(HAS_TRAIT(owner, TRAIT_GIANT))
+		challenger_stamina_mod++
+	if(HAS_TRAIT(owner, TRAIT_DWARF))
+		owner_stamina_mod++
+
+	if(HAS_TRAIT(challenger, TRAIT_GIANT))
+		owner_stamina_mod++
+	if(HAS_TRAIT(challenger, TRAIT_DWARF))
+		challenger_stamina_mod++
+
```
Comment (Dennok) [inline code/datums/components/arm_wrassle.dm:135]: What about add here `strength` dna check?

**Comment by Ryll-Ryll** [comment]
Finish later

**Comment by MrDoomBringer** [comment]
u better

**Comment by ATH1909** [comment]
If you do come back to this, you might want to integrate the moodlet for exercising with this somehow, as well as the strength mutation.

**Comment by Iamgoofball** [comment]
@Ryll-Ryll Did something happen? You closed all your PRs and left the servers.

**Pull request closed without merging**